tap_github/repository_streams.py

```diff
@@ -278,6 +278,114 @@ class CommunityProfileStream(GitHubStream):
+    replication_key = "created_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = False
+
+    def get_records(self, context: Optional[dict] = None) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+        Each row emitted should be a dictionary of property names to their values.
+        """
+        if context and context.get("events", None) == 0:
+            self.logger.debug(f"No events detected. Skipping '{self.name}' sync.")
+            return []
+
+        return super().get_records(context)
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        # TODO - We should think about the best approach to handle this. An alternative would be to
+        # do a 'dumb' tap that just keeps the same schemas as GitHub without renaming these
+        # objects to "target_". They are worth keeping, however, as they can be different from
+        # the parent stream, e.g. for fork/parent PR events.
+        row["target_repo"] = row.pop("repo", None)
+        row["target_org"] = row.pop("org", None)
+        return row
+
+    schema = th.PropertiesList(
+        th.Property("id", th.IntegerType),
+        th.Property("type", th.StringType),
+        th.Property("repo", th.StringType),
+        th.Property("org", th.StringType),
+        th.Property("public", th.BooleanType),
+        th.Property("payload", th.StringType),
+        th.Property("_sdc_repository", th.StringType),
+        th.Property("created_at", th.DateTimeType),
+        th.Property("distinct_size", th.IntegerType),
+        th.Property("head", th.StringType),
+        th.Property("push_id", th.IntegerType),
+        th.Property("ref", th.StringType),
+        th.Property("size", th.IntegerType),
+        th.Property(
+            "target_repo",
+            th.ObjectType(
+                th.Property("id", th.StringType),
+                th.Property("name", th.StringType),
+            ),
+        ),
+        th.Property(
+            "target_org",
+            th.ObjectType(
+                th.Property("id", th.StringType),
+                th.Property("login", th.StringType),
+            ),
+        ),
+        th.Property(
+            "actor",
+            th.ObjectType(
+                th.Property("id", th.IntegerType),
+                th.Property("login", th.StringType),
+                th.Property("display_login", th.StringType),
+                th.Property("avatar_url", th.StringType),
+                th.Property("gravatar_id", th.StringType),
+                th.Property("url", th.StringType),
+            ),
+        ),
+        th.Property(
+            "payload",
+            th.ObjectType(
+                th.Property("before", th.StringType),
+                th.Property("action", th.StringType),
+                th.Property("comment", th.StringType),
+                th.Property("issue", th.StringType),
```

Comment:
at least on some events, `payload.issue` (and `payload.comment`) is a full-blown object, not just a string, for instance on `IssueCommentEvent`. Will this just stringify the content?
Same with `review` on `PullRequestReviewEvent` for instance.

Reply:
@aaronsteers do you know what the default behavior is in this case? Or what you would recommend doing instead?

Reply:
@laurentS @ericboucher 

> Will this just stringify the content?

I don't think so 🤔. It will just output either a string or an object, so targets downstream may fail during record-schema validation if an object is sent.

If the two record possibilities are

```json
{
  "issue": "<some-issue-id>",
  ...
}
```

```json
{
  "issue": {
    "id": "<some-issue-id>",
    ...
  },
  ...
}
```

then I'd recommend unifying them during `post_process` and declare them as a `th.ObjectType()`.

One option if the `object` overlaps with the `string` in on know  is to make this as `th.ObjectType()`, with

Reply:
I expended the objects and I think the PR is ready for a final review.

@aaronsteers @edgarrmondragon are there any tools to extract the schema/typing directcly from an OpenAPI file? It would be pretty neat if we could simply plug into github/rest-api-description@main/descriptions-next/api.github.com/api.github.com.json to generate and validate types

Reply:
@ericboucher not any that do it as we need out-of-the-box, but I have an implementation here: https://github.com/edgarrmondragon/tap-shortcut/blob/99bb738608305baa11fe26acfe32d2c9c7089a58/tap_shortcut/tap.py#L86-L121.

Wdyt?

Reply:
I'd love to see if the OpenAPI spec can speed up SDK tap development. I've dropped an issue here: https://gitlab.com/meltano/sdk/-/issues/310

@ericboucher - Do you mind leaving a quick comment on that issue with your thoughts?